{"name": "OOP: 8.4.14", "url": "https://stepik.org/lesson/808351/step/14", "tests": [{"id": 0, "input": "@exception_decorator\ndef func(x):\n    return 2*x + 1\n    \nprint(func(1))\nprint(func('bee'))", "output": "(3, None)\n(None, <class 'TypeError'>)"}, {"id": 1, "input": "@exception_decorator\ndef f(x, y):\n    return x * y\n    \nprint(f('stepik', 10))", "output": "('stepikstepikstepikstepikstepikstepikstepikstepikstepikstepik', None)"}, {"id": 2, "input": "@exception_decorator\ndef f(x, y):\n    return x * y\n    \nprint(f('stepik', 'stepik'))", "output": "(None, <class 'TypeError'>)"}, {"id": 3, "input": "@exception_decorator\ndef f(*args, **kwargs):\n    return sum(args) + sum(kwargs.values())\n    \nprint(f(1, 2, 3, param1=4, param2=10))", "output": "(20, None)"}, {"id": 4, "input": "@exception_decorator\ndef f(*args, **kwargs):\n    \"\"\"sum args and kwargs\"\"\"\n    return sum(args) + sum(kwargs.values())\n\n\nprint(f.__name__)\nprint(f.__doc__)\nprint(f(1, 2, 3, param1=4, param2='10'))", "output": "f\nsum args and kwargs\n(None, <class 'TypeError'>)"}, {"id": 5, "input": "sum = exception_decorator(sum)\n\nprint(sum(['199', '1', 187]))", "output": "(None, <class 'TypeError'>)"}], "interactive": false, "memoryLimit": 256, "timeLimit": 15000, "srcPath": "c:\\Users\\asald\\Documents\\OOP\\Module_8\\Module_8.4\\Module_8.4.14\\problem.py", "group": "OOP"}